mitodl/mitxonline
# fix bug on renderCourseInfoCard and add migration to backfill certificate enrollments

# What are the relevant tickets?




https://github.com/mitodl/mitxonline/issues/1643

# Description (What does it do?)

 - Fixing bug on renderCourseInfoCard on program drawer. 
      - Current behavior: If a user has enrolled multiple enrollments for the course and has earned a certificate, it displays the most recent enrollment for the course, not necessarily the one with a certificate. This caused inconsistency between the dashboard and program record because user has no way to view the certificate on the program dashboard but then it says "certificate earned" on the program record page
      -  Expected behavior: It should always display the course card with a certificate link and course passed tag if user has earned a certificate for any run. If none of the enrollments has a certificate, it defaults to the most recent enrollment which is the current behavior.
 -  Adding a migration to backfill the missing enrollments for certificates. 
    -  There are 98 in production https://bi.odl.mit.edu/queries/1290, these are all old DEDP certificates

# How can this be tested?

- Setup course, and program if not exist
- Add courserun enrollment from admin/courses/courserunenrollment/ for program course
- Add passing grade for the above run from admin/courses/courserungrade/
- Add courserun certificate for the above run from admin/courses/courseruncertificate/. 
   - Note that for this step, you will need to set up certificates in CMS following https://github.com/mitodl/mitxonline#certificates
-  Repeat step 2 to add another course run enrollment for the same course, but this time don't create a grade or certificate for it
-  Go to your dashboard to view your program course, you should see something like this
    
![image](https://github.com/mitodl/mitxonline/assets/3138890/ab464240-2e60-473d-baf1-6863ad6c4491)

This should be consistent with the program record 
![image](https://github.com/mitodl/mitxonline/assets/3138890/cada5dd7-2822-4d68-8a04-5ee8143a3ed3)
